bitcoin/src/taproot/mod.rs

```diff
@@ -1067,18 +1070,18 @@ impl<'leaf> ScriptLeaf<'leaf> {
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
-pub struct ControlBlock {
+pub struct ControlBlock<B: AsRef<TaprootMerkleBranch>> {
```

Comment:
Almost correct but we need `TaprootMerkleBranch` to be unsized for this to work and rename the current `TaprootMerkleBranch` to `TaprootMerkleBranchBuf`